README.md

```diff
@@ -30,6 +30,24 @@ Tapioca makes it easy to work with [Sorbet](https://sorbet.org) in your codebase
   * Find useless definitions in shim RBI files from Sorbet's embedded RBI for core and stdlib
   * Synchronization validation for your CI
 
+## Table of Contents
+
+* [Installation](#installation)
+* [Getting Started](#getting-started)
+* [Usage](#usage)
+  * [Generating RBI files for gems](#generating-rbi-files-for-gems)
+  * [Manually requiring parts of a gem](#manually-requiring-parts-of-a-gem)
+  * [Excluding a gem from RBI generation](#excluding-a-gem-from-rbi-generation)
+  * [Changing the strictness level of the RBI for a gem](#changing-the-strictness-level-of-the-rbi-for-a-gem)
+  * [Keeping RBI files for gem up-to-date](#keeping-rbi-files-for-gems-up-to-date)
+  * [Pulling RBI annotations from remote sources](#pulling-rbi-annotations-from-remote-sources)
+  * [Generating RBI files for Rails and other DSLs](#generating-rbi-files-for-rails-and-other-dsls)
+  * [RBI files for missing constants and methods](#rbi-files-for-missing-constants-and-methods)
+  * [Configuration](#configuration)
+  * [Migrating to Tapioca](https://github.com/Shopify/tapioca/wiki/Migrating-to-Tapioca)
+* [Contributing](#contributing)
+* [License](#license)
```

Comment:
This should be generated or we're going to forget to update it.

We already have some code to do related things under https://github.com/Shopify/tapioca/blob/main/tasks/readme.rake. We could use a markdown parser to extract the  headings from the markdown.